src/definations/sheet.ts

```diff
@@ -0,0 +1,154 @@
+import { CellKey, ColumnKey, RowKey } from "./keyTypes.ts";
+import Cell from "./cell.ts";
+import Column from "./column.ts";
+import Row from "./row.ts";
+
+export default class Sheet {
+  defaultStyle: any;
+  settings: any;
+  cell_data: Map<CellKey, Cell>;
+  rows: Map<RowKey, Row>;
+  columns: Map<ColumnKey, Column>;
+  rowPositions: Map<number, RowKey>;
+  columnPositions: Map<number, ColumnKey>;
+
+  default_width: number;
+  default_height: number;
+
+  constructor() {
+    this.defaultStyle = null;
+    this.settings = null;
+    this.cell_data = new Map<CellKey, Cell>();
+    this.rows = new Map<RowKey, Row>();
+    this.columns = new Map<ColumnKey, Column>();
+
+    this.rowPositions = new Map<number, RowKey>();
+    this.columnPositions = new Map<number, ColumnKey>();
+
+    this.default_width = 40;
+    this.default_height = 20;
+  }
+
+  setCellAt(
+    colKey: number,
+    rowKey: number,
+    value: string,
+  ): { rowKey: RowKey; columnKey: ColumnKey; cellKey: CellKey } {
+    const position = this.initializePosition(colKey, rowKey);
+    const row = this.rows.get(position.rowKey);
+    const column = this.columns.get(position.columnKey);
+
+    if (!row || !column)
+      throw new Error(`Failed to set cell at col: ${column} row: ${row}.`);
+
+    return this.setCell(column, row, value);
+  }
+
+  setCell(
+    column: Column,
+    row: Row,
+    value: string,
+  ): { rowKey: RowKey; columnKey: ColumnKey; cellKey: CellKey } {
+    let cell = this.getCell(row.key, column.key);
+    if (!cell) {
+      cell = this.createCell(column, row, value);
+    }
+
+    return { rowKey: row.key, columnKey: column.key, cellKey: cell.key };
+  }
```

Comment:
this method is throwing error on updating cell @Fluglow

Reply:
It looks like Typescript Maps use object identity for keys instead of a hash function we could override. This makes instantiating pre-existing keys, like we're doing in main::setCell, not work for lookups. I'll try to think of a solution.